src/System.IO.Compression/src/System/IO/Compression/DeflateManaged/HuffmanTree.cs

```diff
@@ -2,93 +2,79 @@
-        private short[] _right;
-        private byte[] _codeLengthArray;
+        private readonly int _tableBits;
+        private readonly short[] _table;
+        private readonly short[] _left;
+        private readonly short[] _right;
+        private readonly byte[] _codeLengthArray;
 #if DEBUG
         private uint[] _codeArrayDebug;
 #endif
 
-        private int _tableMask;
+        private readonly int _tableMask;
 
         // huffman tree for static block
-        private static HuffmanTree s_staticLiteralLengthTree;
-        private static HuffmanTree s_staticDistanceTree;
-
-        static HuffmanTree()
-        {
-            // construct the static literal tree and distance tree
-            s_staticLiteralLengthTree = new HuffmanTree(GetStaticLiteralTreeLength());
-            s_staticDistanceTree = new HuffmanTree(GetStaticDistanceTreeLength());
-        }
+        public static HuffmanTree StaticLiteralLengthTree { get; } = new HuffmanTree(GetStaticLiteralTreeLength());
 
-        public static HuffmanTree StaticLiteralLengthTree
-        {
-            get
-            {
-                return s_staticLiteralLengthTree;
-            }
-        }
-
-        public static HuffmanTree StaticDistanceTree
-        {
-            get
-            {
-                return s_staticDistanceTree;
-            }
-        }
+        public static HuffmanTree StaticDistanceTree { get; } = new HuffmanTree(GetStaticDistanceTreeLength());
 
         public HuffmanTree(byte[] codeLengths)
         {
-            Debug.Assert(codeLengths.Length == MaxLiteralTreeElements
-                          || codeLengths.Length == MaxDistTreeElements
-                          || codeLengths.Length == NumberOfCodeLengthTreeElements,
-                          "we only expect three kinds of Length here");
+            Debug.Assert(
+                codeLengths.Length == MaxLiteralTreeElements ||
+                codeLengths.Length == MaxDistTreeElements ||
+                codeLengths.Length == NumberOfCodeLengthTreeElements,
+                "we only expect three kinds of Length here");
             _codeLengthArray = codeLengths;
 
             if (_codeLengthArray.Length == MaxLiteralTreeElements)
-            {  // bits for Literal/Length tree table
+            {
+                // bits for Literal/Length tree table
                 _tableBits = 9;
             }
             else
-            {          // bits for distance tree table and code length tree table
+            {
+                // bits for distance tree table and code length tree table
                 _tableBits = 7;
             }
             _tableMask = (1 << _tableBits) - 1;
 
+            _table = new short[1 << _tableBits];
+
+            // I need to find proof that left and right array will always be
+            // enough. I think they are.
+            _left = new short[2 * _codeLengthArray.Length];
+            _right = new short[2 * _codeLengthArray.Length];
```

Comment:
I wanted to make the fields `readonly`. I considered moving the entire contents of `CreateTable` inside the constructor, but in the end, just moved the field initialization to the constructor and left the rest in `CreateTable`.